core/src/main/java/org/apache/iceberg/util/ManifestFileUtil.java

```diff
@@ -53,6 +55,10 @@ private static class FieldSummary<T> {
     }
 
     boolean canContain(Object value) {
+      if (Types.UnknownType.get().equals(type)) {
```

Comment:
Should this be moved after the null and nan value checks? The null and nan counts are still valid, right?

This should also have a test update.